n3fit/src/n3fit/model_trainer.py

```diff
@@ -853,7 +862,12 @@ def hyperparametrizable(self, params):
         exp_models = []
 
         # Generate the grid in x, note this is the same for all partitions
-        xinput = self._xgrid_generation()
+        xinput, inputs_unique = self._xgrid_generation()
+        
+        # compute photon:
+        photon=Photon(theoryid=self.theoryid, fiatlux_runcard=self.fiatlux_runcard)
+        photon_array = np.concatenate([photon.photon_fitting_scale(xgrid[0]) for xgrid in inputs_unique])
```

Comment:
```suggestion
        photon_array = photon.photon_fitting_scale(xinput.input_l.tensor_content)
```

(see above, note the point about the change of name for `input_l`.

Also, since there might be situations in which more than one replica is fitted at once, you might want:

```python
full_photon = []
for xgrid in xinput.input_l.tensor_content:
    ph = photon.photon_fitting_scale(xgrid.squeeze())
    full_photon.append(ph)
photon_array = np.stack(full_photon)
```

Reply:
The reason I implemented it this way is that ```xinput.input_l.tensor_content``` contains the concatenation of all the xgrids and therefore the final grid is not ordered. Since the computation of the photon requires the calling of ```EKO``` we need to pass a grid that is ordered otherwise ```EKO``` will crash (at least I guess since I never tried to call it with such grid).

Reply:
But the single arrays are not guaranteed to be ordered. They might be, by chance, but it is not a requirement.

If you really need the arrays to be ordered then you should take it unordered and then order it internally inside photon_fitting_scale.